default.nix

```diff
@@ -0,0 +1,251 @@
+      checksumTable2
+      checksumWorksheet
+    ];
+  };
+
+  dependencyContentRecur = content: builtins.concatMap
+    (item: (dependencyContentRecur item.dependencies) ++ [ item.content ])
+    content;
+  dependencyContent = pages: lib.lists.unique (
+    (map (dep: dep.content) standardDependencies) ++
+    (builtins.concatMap (page: dependencyContentRecur page.dependencies) pages)
+  );
+
+  renderBooklet = booklet:
+    let
+      addPage = content: pageData: {
+        content = content.content + lib.optionalString (pageData ? sourceHeader) ''
+          %****************************************************************
+          %*
+          %* ${pageData.sourceHeader}
+          %*
+          %****************************************************************
+        '' + ''
+          %%Page: ${toString content.nextPgIdx} ${toString content.nextPgIdx}
+          ${lib.optionalString (pageData ? isLandscape) "%%PageOrientation: Landscape\n"}
+          %%BeginPageSetup
+          /pgsave save def
+          %%EndPageSetup
+          ${if pageData ? isLandscape then "landscapePage" else "portraitPage"} begin
+          ${lib.optionalString (pageData ? drawFooter) "${toString content.nextFooterIdx} drawFooter"}
+          ${pageData.content}
+          end
+          pgsave restore
+          showpage
+        '';
+        nextFooterIdx = content.nextFooterIdx + (if pageData ? drawFooter then 1 else 0);
+        nextPgIdx = content.nextPgIdx + 1;
+      };
+      initialContent = {
+        content = ''
+          %!PS-Adobe-3.0
+          %%Orientation: Portrait
+          %%Pages: ${toString (builtins.length booklet.pages)}
+          %%EndComments
+          %%BeginSetup
+          ${toString (dependencyContent (booklet.pages))}%%EndSetup
+
+          %************************************************************************
+          %************************************************************************
+          %*
+          %* Section Three: Page Rendering
+          %*
+          %************************************************************************
+          %************************************************************************
+
+        '';
+        nextPgIdx = 1;
+        nextFooterIdx = 1;
+      };
+      finalContent = builtins.foldl' addPage initialContent booklet.pages;
+    in
+    pkgs.writeTextFile {
+      name = booklet.name;
+      text = finalContent.content + ''
+        %%EOF
+      '';
+    };
+  checkSinglePage = page: renderBooklet {
+    name = "test-single-page.ps";
+    pages = [ page ];
+  };
+in
+stdenv.mkDerivation {
+  name = "codex32${shortId}";
+
+  nativeBuildInputs = if doPdfGeneration then [ ghostscript ] else [ ];
+
+  phases = [ "buildPhase" ] ++ lib.optionals doCheck [ "checkPhase" ];
+
+  buildPhase = ''
+    set -e
+
+    FULL_BW="${renderBooklet fullBooklet}"
+
+    # Copy output Postscript into place
+    mkdir "$out"
+    cd "$out"
+    ln -s "$FULL_BW" SSS32.ps
```

Comment:
You changed this from `cp` to `ln -s`, but below you patch the file with `sed`.  Does this even work?  the content of `FULL_BW` should be an immutable nix-store file, so I don't think you can patch it.

Reply:
Should renderBooklet take a revision argument?

Reply:
No, it does not work. The `sed` line appears to be doing nothing, and since it does nothing, does not even try to write to the file, and returns success.

As for "should renderBooklet just take a `revision` argument?" let me investigate that. Would be more elegant than rendering the book and then patching it after-the-fact, but might make templating a bit weird.

Reply:
Oh, `sed -i` will replace linked files unless you pass it `--follow-symlinks`.

Linux.